test/gurka/test_route.cc

```diff
@@ -526,6 +526,7 @@ TEST(Standalone, BridgingEdgeIsRestricted) {
 TEST(Standalone, AvoidExtraDetours) {
   // Check that we don't take extra detours to get the target point. Here is
   // a special usecase that breaks previous logic about connection edges in bidirectional astar.
+  // TODO(nils): this currently fails because both trees hit a not_thru area on DE (fwd) and DC(rev)
```

Comment:
so in the map below, the route wants to from 1 -> 2 with bidir a*. the test sets D -> A edges and D -> F edges as `not_thru`, which means neither can the fwd search see the rev tree, nor can the rev search see any of the fwd tree and they both expand into the HIJK region, taking a massive detour.

I _think_ it fails now because we don't even record `not_thru` edges which means a connection on those can't happen anymore.

2 things:
1. this test describes a real failure mapbox noticed on their master a while back, but jeez this is kinda rare.. and not sure it really justifies the extra complex logic that was there before
2. but ok, it shows a rare bug. however, the real problem here is that we _do_ find a connection, it's just a huge detour. if we wouldn't find a connection in the first pass, that's fine, we would find it on a second pass. anyways, obviously it takes `CD` & `DE` edges in either case. in master it'd make that transition directly to have the path `{ "AB", "BC", "CD", "DE", "EF" }`. in this PR, it can't and instead `D` loops back on itself, i.e. with path `{ "AB", "BC", "CD", "DH", "HI", "IJ", "JK", "KH", "DH", "DE", "EF" }` which simply adds the sub-path `"DH", "HI", "IJ", "JK", "KH", "DH"`. now what I wonder: couldn't we detect that while recosting or in triplegbuilder and just cut that completely superfluous part out of the final path?

Reply:
actually I'm pretty sure this could fairly easily be done. the only thing to be wary about that I can think of are turn restrictions. if in the case above `CD` -> `DE` was restricted, we'd allow the detour, otherwise we'd cut the superfluous part and recost the remainder of the path accordingly. it'll need to be recorded while `recost_forward` is running, I'll try an implementation later. 

it's only interesting for bidirectional algorithms. and while it'll be a bit more of a burden since we'd expand that detour where we wouldn't have to, this should be a very rare case and would maybe help other cases of unnecessary detours as well?!